cas/src/test/java/org/springframework/security/cas/jackson2/CasAuthenticationTokenMixinTest.java

```diff
@@ -0,0 +1,128 @@
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.springframework.security.cas.authentication.CasAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.jackson2.SimpleGrantedAuthorityMixin;
+import org.springframework.security.jackson2.UnmodifiableSetMixin;
+import org.springframework.security.jackson2.UserMixin;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * @author Jitendra Singh
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CasAuthenticationTokenMixinTest {
+
+	ObjectMapper buildObjectMapper() {
+		ObjectMapper mapper = new ObjectMapper()
+				.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
+		mapper.setVisibilityChecker(mapper.getVisibilityChecker().withVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY));
+		mapper.addMixIn(CasAuthenticationToken.class, CasAuthenticationTokenMixin.class)
+				.addMixIn(SimpleGrantedAuthority.class, SimpleGrantedAuthorityMixin.class)
+				.addMixIn(Collections.unmodifiableSet(Collections.EMPTY_SET).getClass(), UnmodifiableSetMixin.class)
+				.addMixIn(User.class, UserMixin.class)
+				.addMixIn(AssertionImpl.class, AssertionImplMixin.class)
+				.addMixIn(AttributePrincipalImpl.class, AttributePrincipalImplMixin.class);
+		return mapper;
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void nullKeyTest() {
+		new CasAuthenticationToken(null, "user", "pass", Collections.<GrantedAuthority>emptyList(),
+				new User("user", "pass", Collections.<GrantedAuthority>emptyList()), null);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void blankKeyTest() {
+		new CasAuthenticationToken("", "user", "pass", Collections.<GrantedAuthority>emptyList(),
+				new User("user", "pass", Collections.<GrantedAuthority>emptyList()), null);
+	}
+
+	@Test
+	public void serializeCasAuthenticationTest() throws JsonProcessingException, JSONException {
+		String key = "casKey";
+		Date startDate = new Date();
+		Date endDate = new Date();
+		Collection<? extends GrantedAuthority> authorities = Collections.singletonList(new SimpleGrantedAuthority("ROLE_USER"));
+		Assertion assertion = new AssertionImpl(new AttributePrincipalImpl("assertName"), startDate, endDate, startDate, Collections.<String, Object>emptyMap());
+		CasAuthenticationToken token = new CasAuthenticationToken(key, "user", "pass", authorities,
+				new User("user", "pass", authorities), assertion);
+
+		String expectedJson = "{\"@class\": \"org.springframework.security.cas.authentication.CasAuthenticationToken\", \"keyHash\": "+key.hashCode()+"," +
```

Comment:
It would be good to place the JSON and `CasAuthenticationToken` in a member variable. This would ensure that what is serialized could easily be deserialized.